workos/utils/request.py

```diff
@@ -45,6 +45,9 @@ def set_base_api_url(self, base_api_url):
     def generate_api_url(self, path):
         return self.base_api_url.format(path)
 
+    def build_parameterized_url(self, url, **params):
+        return url.format(**params)
```

Comment:
While I wonder whether we can just use formatted string literals, one reason to prefer a helper like this would be to do automatic URL component escaping:

```py
def build_parameterized_url(self, url, **params):
    escaped_params = {k: urllib.parse.quote(str(v)) for k, v in params.items()}
    return url.format(**escaped_params)
```

Reply:
@mthadley The initial idea that triggered the addition of this function was so we could test these interpolations. A few days ago a customer reported a bug caused by the interpolation silently failing.

Now that you bring up the escaping, that's an additional support we could have with this method. Thanks for bringing this up! I committed the suggestion.